packages/workspace/src/generators/convert-to-buildable/schema.json

```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "$id": "SchematicsConvertToBuildable",
+  "title": "Convert library to be a buildable lib",
+  "type": "object",
+  "cli": "nx",
+  "examples": [
+    {
+      "command": "g @nrwl/workspace:convert-to-buildable --project my-feature-lib",
+      "description": "Convert the my-feature-lib project to be buildable"
+    }
+  ],
+  "properties": {
+    "project": {
+      "description": "Name of project to convert to buildable lib.",
+      "type": "string"
+    },
+    "skipFormat": {
+      "description": "Skip formatting files.",
+      "type": "boolean",
+      "default": false
+    },
+    "type": {
+      "description": "The type of library that this is.",
+      "type": "string",
+      "enum": ["js", "node", "nest", "next", "react", "detox", "web"]
```

Comment:
Ideally, I think this would be built into each package:

`nx g @nrwl/workspace:library lib1 --buildable` would be equivalent to say like `nx g @nrwl/workspace:library lib1 && nx g @nrwl/workspace:add-build lib1`

1. Having a switch statement here effectively means you're writing 7 generators in 1
2. You use these generators to add the target configuration to any project similar to how `@nrwl/jest:jest-project` adds a `test` target onto an existing project.
3. `@nrwl/react:add-build` would inherit `@nrwl/web:add-build`
4. We already have the logic in our library generators, and only need to extract it to be reusable.

We have 4 executors (3 in this PR + angular) so I think this would probably wind up being 4 executors?

We used this kind of strategy for `convert-to-eslint`.

What do you all think?

Reply:
Yeah it might make more sense to have this split into each package. it would help keep like-minded code together, especially if we ever need to change how we create buildable libs and their config (e.g. angular might have to have some work done to the buildable lib implementation).

If the functionality to generate a buildable lib was extracted into a generator `add-build` that generator could just be reused in the library generator when `--buildable` is passed, as you suggested.

I like it, I think that makes more sense.

We  could still have a generator in workspace package that will then call the underlying generators.

Reply:
If we have these split into their respective packages, and have the root level 'add-buildable --all' option, how would it know run the generators? Should a generator in @nrwl/workspace rely on generators in other verticals? I wouldn't think so, but maybe dynamic imports would be a "good enough" way of doing them without adding deps to each plugin that supports buildable libs.